hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/workflows/HandleException.java

```diff
@@ -74,6 +77,8 @@ public static void validateFalse(final boolean flag, final ResponseCodeEnum erro
      * @param cause the {@link Throwable} that caused this exception.  This is ignored.
      * @throws UnsupportedOperationException always.  This constructor must not be called.
      */
+    // Suppressing the warning that the constructor and arguments are not used
+    @SuppressWarnings({"java:S1144", "java:S1172"})
     private HandleException(@NonNull final ResponseCodeEnum responseCode, @Nullable final Throwable cause) {
         throw new UnsupportedOperationException("HandleException must not chain a cause");
     }
```

Comment:
Does anyone know the idea behind this constructor? What is it used for?

Reply:
The idea was to prevent anybody from providing a cause to a `HandleException` as it will be ignored. Personally, I find the above method overly cautious, but then again, it does not really matter. 🙂